extension/lib/shopify/CollectionRepository.js

```diff
@@ -27,7 +27,18 @@ class ShopifyCollectionRepository {
    * @returns {Promise<ShopifyCollection[]>}
    */
   async list () {
-
+    const rawCollections = await this._shopifyCollectionCommandFactory.createList().execute()
+
+    const productCountPromises = rawCollections.map(collection => {
+      return this._shopifyCollectionCommandFactory.createGetProductCount().execute(collection.id)
+    })
+
+    const collectionCategoryCounts = await Promise.all(productCountPromises)
+    const shopifyCollections = []
+    for (let countIndex in rawCollections) {
```

Comment:
you can use return shopifyCollections.map((rawCollections) => { })

Reply:
if you won't change it you should be careful using this contrstruct. Just check the downside of a for(a in b) loop: https://stackoverflow.com/a/500531
It will walk through all properties and also through not existing Indizes.